src/main/kotlin/org/opensearch/commons/alerting/model/Comment.kt

```diff
@@ -0,0 +1,159 @@
+package org.opensearch.commons.alerting.model
+
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._ID
+import org.opensearch.commons.alerting.util.instant
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.alerting.util.optionalUserField
+import org.opensearch.commons.alerting.util.optionalUsernameField
+import org.opensearch.commons.authuser.User
+import org.opensearch.core.common.io.stream.StreamInput
+import org.opensearch.core.common.io.stream.StreamOutput
+import org.opensearch.core.common.io.stream.Writeable
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.XContentBuilder
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+import java.time.Instant
+
+data class Comment(
+    val id: String = NO_ID,
+    val entityId: String = NO_ID,
+    val content: String,
+    val createdTime: Instant,
+    val lastUpdatedTime: Instant?,
+    val user: User?
```

Comment:
will this be optional field?

Reply:
`lastUpdatedTime` will be an optional field because upon a Comment's creation, it will not have a last updated time, only a created time. `lastUpdatedTime` is only assigned when a User makes a subsequent call to the same Comment to edit/update it. `user` will also be an optional field because there's no guarantee that the Security Plugin will be installed.